Makes it EVEN EASIER to work with atom item interactions ft. "Leaf and Branch" & "Death to Chains"

MrMelbert opened the pull request:
## About The Pull Request

When I made the new item interaction chain, I did this pattern for items interacting with atoms: 

https://github.com/tgstation/tgstation/blob/d2956ae821c6a865d8ee3de0d4b92979a9d08e32/code/game/atom/atom_tool_acts.dm#L85-L86

This is VERY clean and makes it ridiculously easy to add item-to-atom interactions without worry about needing to fuss around with calling parent and checking parent return value, IE, dealing with the dreaded chain. 

Of course for some reason when I did this I didn't do the same for atom-from-item interactions themselves, relying on people call parent and check for blockers. Which worked in my head but is not easy to pick up as a new contributor. 

So here we are, hopefully the final version of the atom-item-non-combat-interact chain. 

Base `item_interaction` that handles all the signals and calling relevant procs is now `base_item_interaction`

`item_interaction` is now a stub proc that atoms can override at whim without worrying about parent calls (unless subtypes implement it). 

This results in MUCH cleaner and easier to work with code. 

And as an added bonus, it's (technically) completely backwards compatible with existing code. I changed it just for posterity but for downstreams and such, they don't even need to change anything, and it'll work as it did before. 

## Changelog

:cl: Melbert
refactor: Atom-Item interactions have been refactored once more. Report any oddities. 
/:cl:

MrMelbert committed:
Makes it even easier to work with atom item interactions through the power of branching

MrMelbert committed:
Foam tweak

MrMelbert committed:
Deletes unused code

MrMelbert committed:
How did that get there

SyncIt21 commented on code/modules/reagents/chemistry/machinery/chem_mass_spec.dm:
```diff
@@ -149,7 +149,7 @@
 
 /obj/machinery/chem_mass_spec/item_interaction(mob/living/user, obj/item/item, list/modifiers, is_right_clicking)
```
```suggestion
/obj/machinery/chem_mass_spec/item_interaction(mob/living/user, obj/item/item, list/modifiers)
```

Missed one

MrMelbert committed:
Deletes some unused args

MrMelbert committed:
Used Unused

MrMelbert committed:
Merge branch 'master' into we_branching_fr

jlsnow301 commented on code/game/objects/structures/tables_racks.dm:
```diff
@@ -858,12 +858,12 @@
+/obj/structure/rack/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	if((tool.item_flags & ABSTRACT) || user.combat_mode)
+		return NONE
 	if(user.transferItemToLoc(tool, drop_location(), silent = FALSE))
 		return ITEM_INTERACT_SUCCESS
+	return NONE
```
Does this mean we'll hit it if transfer fails? Can this be made into a guard instead?

MrMelbert replied:
I suppose we can
```suggestion
	if(user.transferItemToLoc(tool, drop_location(), silent = FALSE))
		return ITEM_INTERACT_SUCCESS
	return ITEM_INTERACT_BLOCKING
```

MrMelbert committed:
Change to hit

MrMelbert committed:
Fixes machine frames.
Item interaction now runs before all the tool_acts which means this interaction, which attempts to add ANY item to the machine, always blocks.

MrMelbert committed:
Merge branch 'we_branching_fr' of https://github.com/MrMelbert/tgstation into we_branching_fr

Ketrai commented:
Just as an FYI this broke the tray interactions with ovens introduced in #82566

Ketrai commented:
It's only the interaction when left clicking an oven with a tray.

I cannot for the life of me figure out why the proc isn't working, debugging right now.

Ketrai commented:
#82787 Fixed
